src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -3477,4 +3477,36 @@ WindowID MessageHelper::ExtractWindowIdFromSmartObject(
   return mobile_apis::PredefinedWindows::DEFAULT_WINDOW;
 }
 
+void MessageHelper::RemoveEmptyMessageParams(
+    smart_objects::SmartObject& msg_params) {
+  if (msg_params.empty()) {
+    return;
+  }
+
+  for (const auto& item : msg_params.enumerate()) {
+    auto& param = msg_params[item];
+
+    if (param.empty()) {
+      msg_params.erase(item);
+      break;
+    }
+
+    if (smart_objects::SmartType_Array == param.getType()) {
+      auto& array = *param.asArray();
```

Comment:
Please check that this pointer is not null, and type of auto& array is unclear. What is that?

Reply:
I changed auto on ns_smart_device_link::ns_smart_objects::SmartArray& array

Reply:
But I don't see any changes. May you push your changes?

Reply:
And please look at following:

```
SmartArray* SmartObject::asArray() const {
  if (m_type != SmartType_Array) {
    return NULL;
  }
  return m_data.array_value;
}
```
I think that *param.asArray() may be unsafe because we get pointer here. So I propose to make sure that result of execution asArray() method is not null, and only after that dereference it. I don't propose to use `ns_smart_device_link::ns_smart_objects::SmartArray&` array name, it's too long.